x-pack/plugins/lens/public/app_plugin/shared/edit_on_the_fly/lens_configuration_flyout.tsx

```diff
@@ -61,66 +57,37 @@ export function LensEditConfigurationFlyout({
   closeFlyout,
   adaptersTables,
 }: EditConfigPanelProps) {
-  const currentDataViewId = dataView.id ?? '';
   const datasourceState = attributes.state.datasourceStates[datasourceId];
   const activeVisualization = visualizationMap[attributes.visualizationType];
   const activeDatasource = datasourceMap[datasourceId];
-  const dispatchLens = useLensDispatch();
   const { euiTheme } = useEuiTheme();
-  const dataViews = useMemo(() => {
-    return {
-      indexPatterns: {
-        [currentDataViewId]: dataView,
-      },
-      indexPatternRefs: [],
-    } as unknown as DataViewsState;
-  }, [currentDataViewId, dataView]);
-  dispatchLens(
-    updateStateFromSuggestion({
-      newDatasourceId: datasourceId,
-      visualizationId: activeVisualization.id,
-      visualizationState: attributes.state.visualization,
-      datasourceState,
-      dataViews,
-    })
-  );
 
-  const datasourceLayers: DatasourceLayers = useMemo(() => {
-    return {};
-  }, []);
   const activeData: Record<string, Datatable> = useMemo(() => {
     return {};
   }, []);
   const layers = activeDatasource.getLayers(datasourceState);
   layers.forEach((layer) => {
-    datasourceLayers[layer] = datasourceMap[datasourceId].getPublicAPI({
-      state: datasourceState,
-      layerId: layer,
-      indexPatterns: dataViews.indexPatterns,
-    });
     if (adaptersTables) {
       activeData[layer] = Object.values(adaptersTables)[0];
     }
   });
 
-  const dateRange = getResolvedDateRange(startDependencies.data.query.timefilter.timefilter);
-  const framePublicAPI = useMemo(() => {
-    return {
-      activeData,
-      dataViews,
-      datasourceLayers,
-      dateRange,
-    };
-  }, [activeData, dataViews, datasourceLayers, dateRange]);
+  const framePublicAPI = useLensSelector((state) => selectFramePublicAPI(state, datasourceMap));
+  const { isLoading } = useLensSelector((state) => state.lens);
+  if (isLoading) return null;
+  const framePublicAPIEnhanced = {
+    ...framePublicAPI,
+    activeData,
```

Comment:
why do you have to pass `activeData` from here instead of using the one the selector `selectFramePublicAPI` gives you? 🤔

Reply:
I have possibly missed that Marta, let me check it.

Reply:
Done Marta! https://github.com/elastic/kibana/pull/161256/commits/6a9b601356d51ff983ccad90adf3edb9ee7fced2